### PR TITLE
ECS progressive deployment implementation

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Support progressive deployment with ECS cloud provider
```

### CODE REVIEW

File: pkg/app/piped/executor/ecs/rollback.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func rollback(ctx context.Context, in *executor.Input, cloudProviderName string,
 
 	// Remove old taskSet if existed.
 	if prevPrimaryTaskSet != nil {
-		if err = client.DeleteTaskSet(ctx, *service, *prevPrimaryTaskSet); err != nil {
+		if err = client.DeleteTaskSet(ctx, *service, *prevPrimaryTaskSet.TaskSetArn); err != nil {
 			in.LogPersister.Errorf("Faield to remove unused taskSet %s: %v", *prevPrimaryTaskSet.TaskSetArn, err)
```
Comment (nghialv) [inline pkg/app/piped/executor/ecs/rollback.go:144]: nit: "Failed"

**Comment by nghialv** [comment]
Just found a typo.
/lgtm

File: pkg/app/piped/executor/ecs/rollback.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/app/piped/executor/ecs/rollback.go]: Thanks 🙏

**Pull request merged**